### ISSUE TITLE
The time cost is about 2 seconds

### ISSUE THREAD

**Issue by zhengyi2015mi** [fizyr/keras-retinanet#310]
The time cost is about 2 seconds for 000000008021.jpg when I use the model resnet50_coco_best_v1.2.2.h5 with P40. I check out the tensorflow with GPU version. I want to know why causes so much time cost.

**Comment by hgaiser** [comment]
It could be that your GPU still needs to be initialized with the structures necessary for computation. Try running it a couple (5?) times and see the processing time go down.

**Comment by zhengyi2015mi** [comment]
In ResNet50RetinaNet.ipynb file, my core code is rewritten as:
# process image
start = time.time()
detections = []
couple_time = 100
for x in range(couple_time):
    _, _, detections = model.predict_on_batch(np.expand_dims(image, axis=0))
print("processing time: ", (time.time() - start)/couple_time)
But the average time cost is still about 0.7 seconds.

**Comment by zhengyi2015mi** [comment]
In ResNet50RetinaNet.ipynb file, my core code is rewritten as:

start = time.time()
detections = []
couple_time = 100
for x in range(couple_time):
_, _, detections = model.predict_on_batch(np.expand_dims(image, axis=0))
print("processing time: ", (time.time() - start)/couple_time)

But the average time cost is still about 0.7 seconds.

**Comment by zhengyi2015mi** [comment]
In ResNet50RetinaNet.ipynb file, my core code is rewritten as:

start = time.time()
detections = []
couple_time = 100
for x in range(couple_time):
	_, _, detections = model.predict_on_batch(np.expand_dims(image, axis=0))
print("processing time: ", (time.time() - start)/couple_time)

But the average time cost is still about 0.7 seconds.

**Comment by de-vri-es** [comment]
you should run one or two detections before starting the timing if you want a reliable measurement of the time after those two. The difference between the first few and subsequent detections could easily be around a factor 10.

**Comment by zhengyi2015mi** [comment]
I run ten times before starting the timing,  but it doesn't matter much.

**Comment by de-vri-es** [comment]
What hardware are you using?

**Comment by hgaiser** [comment]
Also, did you install cuDNN ? It could speed things up significantly if you don't use it.

**Comment by zhengyi2015mi** [comment]
My hardware are Tesla P40 with four cores, each core 24GB memory.

**Comment by zhengyi2015mi** [comment]
My computer's cuDNN information is below:

#define CUDNN_MAJOR      5
#define CUDNN_MINOR      1
#define CUDNN_PATCHLEVEL 5
--
#define CUDNN_VERSION    (CUDNN_MAJOR * 1000 + CUDNN_MINOR * 100 + CUDNN_PATCHLEVEL)
#include "driver_types.h"

**Comment by hgaiser** [comment]
I assume you mean you have 4x P40 GPU's? I'm not familiar with the speed of the P40, but 0.7s seems quite high. For reference, a 1080Ti runs an image in approx. 70-80msec. If you look at the individual processing times, instead of the average, what do you see? Could you share those times?

**Comment by zhengyi2015mi** [comment]
Here is the individual processing times:
('processing time: ', 0.6334168910980225)
('processing time: ', 0.6092500686645508)
('processing time: ', 0.6087579727172852)
('processing time: ', 0.5902149677276611)
('processing time: ', 0.6254301071166992)
('processing time: ', 0.6199560165405273)
('processing time: ', 0.6839289665222168)
('processing time: ', 0.63047194480896)
('processing time: ', 0.6317660808563232)
('processing time: ', 0.594843864440918)
('processing time: ', 0.6383650302886963)
('processing time: ', 0.6555569171905518)
('processing time: ', 0.617469072341919)
('processing time: ', 0.6268770694732666)
('processing time: ', 0.6330549716949463)
('processing time: ', 0.6304709911346436)
('processing time: ', 0.627946138381958)
('processing time: ', 0.6074168682098389)
('processing time: ', 0.607429027557373)
('processing time: ', 0.5928781032562256)
('processing time: ', 0.613365888595581)
('processing time: ', 0.6463260650634766)
('processing time: ', 0.6419470310211182)
('processing time: ', 0.5621778964996338)
('processing time: ', 0.6464149951934814)
('processing time: ', 0.6008830070495605)
('processing time: ', 0.6200301647186279)
('processing time: ', 0.5813651084899902)
('processing time: ', 0.6260781288146973)
('processing time: ', 0.624488115310669)
('processing time: ', 0.5950899124145508)
('processing time: ', 0.5970258712768555)
('processing time: ', 0.617110013961792)
('processing time: ', 0.6044378280639648)
('processing time: ', 0.6271510124206543)
('processing time: ', 0.5878000259399414)

**Comment by hgaiser** [comment]
@zhengyi2015mi do you have access to a GTX 1080Ti, Titan X(p) or anything similar? I've never used the P40 so I don't know much about its capabilities.

**Comment by hgaiser** [comment]
Closing due to inactivity.
